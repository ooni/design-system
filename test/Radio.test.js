/* global describe, test, expect */
import React from 'react'
import { renderWithTheme } from './index'
import { RadioButton, RadioGroup } from '../'

describe('RadioButton', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <RadioButton />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with prop checked', () => {
    const { container } = renderWithTheme(
      <RadioButton defaultChecked />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})


describe('RadioGroup', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <RadioGroup>
        <RadioButton />
      </RadioGroup>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
