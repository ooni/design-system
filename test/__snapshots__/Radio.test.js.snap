// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  z-index: -1;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 9999px;
  color: gray;
}

input:checked ~ .c0 {
  color: #0588cb;
}

input:focus ~ .c0 {
  background-color: #0588cb20;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
  width: 24px;
  height: 24px;
}

input:checked ~ .c0 {
  display: block;
}

.c7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 24px;
  height: 24px;
}

input:checked ~ .c0 {
  display: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<label
  class="c0 c1 c2"
>
  <div
    class="c0 c3"
  >
    <input
      class="c0 c4"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="c0 c5 c0 c6"
      fill="currentcolor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="c0 c7 c0 c6"
      fill="currentcolor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
</label>
`;

exports[`RadioButton renders with prop checked 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  z-index: -1;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 9999px;
  color: gray;
}

input:checked ~ .c0 {
  color: #0588cb;
}

input:focus ~ .c0 {
  background-color: #0588cb20;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
  width: 24px;
  height: 24px;
}

input:checked ~ .c0 {
  display: block;
}

.c7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 24px;
  height: 24px;
}

input:checked ~ .c0 {
  display: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<label
  class="c0 c1 c2"
>
  <div
    class="c0 c3"
  >
    <input
      checked=""
      class="c0 c4"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="c0 c5 c0 c6"
      fill="currentcolor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="c0 c7 c0 c6"
      fill="currentcolor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
</label>
`;

exports[`RadioGroup renders 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  z-index: -1;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.c7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 9999px;
  color: gray;
}

input:checked ~ .c0 {
  color: #0588cb;
}

input:focus ~ .c0 {
  background-color: #0588cb20;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
  width: 24px;
  height: 24px;
}

input:checked ~ .c0 {
  display: block;
}

.c8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 24px;
  height: 24px;
}

input:checked ~ .c0 {
  display: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0 c1 c2"
>
  <label
    class="c0 c3 c2"
  >
    <div
      class="c0 c4"
    >
      <input
        checked=""
        class="c0 c5"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="c0 c6 c0 c7"
        fill="currentcolor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c0 c8 c0 c7"
        fill="currentcolor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
  </label>
</div>
`;
